LoopKitUI/CGMManagerUI.swift

```diff
@@ -10,19 +10,27 @@ import SwiftUI
 import LoopKit
 
 public protocol CGMManagerUI: CGMManager, DeviceManagerUI, PreferredGlucoseUnitObserver {
-    /// Provides a view controller for setting up and configuring the manager if needed.
+    /// Create and onboard a new CGM manager.
     ///
-    /// If this method returns nil, it's expected that `init?(rawState: [:])` creates a non-nil manager
-    static func setupViewController(glucoseTintColor: Color, guidanceColors: GuidanceColors) -> (UIViewController & CGMManagerSetupViewController & CompletionNotifying)?
+    /// - Parameters:
+    ///     - colorPalette: Color palette to use for any UI.
+    /// - Returns: Either a conforming view controller to create and onboard the CGM manager, a newly created and onboarded CGM manager, or an error.
+    static func setupViewController(colorPalette: LoopUIColorPalette) -> UIResult<UIViewController & CGMManagerCreateNotifying & CGMManagerOnboardNotifying & CompletionNotifying, CGMManagerUI, Error>
```

Comment:
[nit] Using `CGMManagerUI` here but in OnboardingUI using `CGMManager`. Intentional?

Reply:
Yes, unfortunately. There is a minor API dissonance between PumpManager/CGMManager and PumpManagerUI/CGMManagerUI. Basically, it results from the fact that a pump manager (with UI) can also be a CGM manager (without UI). 

The `setupViewController` is, by definition, always from a UI (since it is declared on the UI protocol). However, when onboarding is instantiating a manager/service it might not have a UI (as is the case of the pump manager without a CGM manager UI).